### PR TITLE
Tentative lhotse --> kaldi manifests conversion for multiple channels 

### PR DESCRIPTION
**Pull request by popcornell**
As title says, works with CHiME6 + DiPCo + Mixer6 Speech. 
But may break in other circumstances.

### CODE REVIEW

**Comment by popcornell** [comment]
NOTE: isort installation fails on my side causing pre-commit hook to fail.

**Comment by desh2608** [comment]
Tagging @jtrmal since he's been working on the Kaldi import/export stuff.

@pzelasko We need this option to support the ESPNet2 baseline for the CHiME-challenge (https://github.com/espnet/espnet/pull/4894). My main concern right now is that the exported manifests for multi-channel recording won't be consistent when imported back into Lhotse, but I think most of the current use-case for this is limited to exporting the manifests, so perhaps we can get this checked in?

**Comment by pzelasko** [comment]
Thanks guys! (and nice too see you back here @popcornell :))

Does this PR introduce any regression? As long as there's no regression for the existing cases, I am OK with incosistencies on importing back.

**Comment by desh2608** [comment]
> Thanks guys! (and nice too see you back here @popcornell :))
> 
> Does this PR introduce any regression? As long as there's no regression for the existing cases, I am OK with incosistencies on importing back.

It shouldn't change anything for the existing single-channel export AFAIK, but let's wait for the tests to confirm that.

**Review by pzelasko — APPROVED** [review]
Cool, LGTM!

**Comment by popcornell** [comment]
@pzelasko Ahaha yeah I am back, amazed on how much lhotse has grown. 

I think consistency is impossible for CHiME-6 like  datasets where you have multi-channel data splitted over multiple single channel files (even for the same device). 
These will be dumped as single channel kaldi entries in wav.scp but they are regarded as multi-channel in lhotse no ?

File: lhotse/kaldi.py [hunk 1]
```diff
@@ -403,10 +468,17 @@ def make_wavscp_channel_string_map(
             # Handles non-WAVE audio formats and multi-channel WAVEs.
             audios = dict()
             for channel in source.channels:
-                audios[channel] = (
-                    f"ffmpeg -threads 1 -i {source.source} -ar {sampling_rate} "
-                    f"-map_channel 0.0.{channel}  -f wav -threads 1 pipe:1 |"
-                )
+                if len(source.channels) == 1:
```
[the remark this answers is not in the document]
Comment (desh2608) [reply lhotse/kaldi.py:471]: It might be worth adding a comment here to point out limitations. Something like `NOTE (@popcornelle): ....`

**Comment by pzelasko** [comment]
> I think consistency is impossible for CHiME-6 like datasets where you have multi-channel data splitted over multiple single channel files (even for the same device). These will be dumped as single channel kaldi entries in wav.scp but they are regarded as multi-channel in lhotse no ?

That's right, Kaldi data directories don't really "know" about multi-channel recordings, we could probably work around it by adding some convention like special recording-id suffixes that help group relevant wav.scp lines back into a single Recording, but if nobody needs it today, there's likely no point.

BTW don't worry about failing test for python 3.9, it's an RNG flake, we can merge it anyway once the rest finishes.

**Comment by jtrmal** [comment]
As long as tests finish, I'm fine with it. The only long-term issue will be
if there are different conventions for multichannel audio in kaldi recipes
audio files, but let's not worry about that for now
y.

**Comment by desh2608** [comment]
@popcornell is this ready to merge?

**Comment by jtrmal** [comment]
WOuld be great if you guys could add tests testing the multi-channel setup
y.

File: test/test_kaldi_dirs.py [hunk 2]
```diff
@@ -120,13 +120,12 @@ def multi_channel_kaldi_dir():
...
         "segments": {
-            "lbi-1272-135031-0000-A": "lbi-1272-135031-0000_0 0.0 10.885",
-            "lbi-1272-135031-0000-B": "lbi-1272-135031-0000_1 0.0 10.885",
+            "lbi-1272-135031-0000-A-0": "lbi-1272-135031-0000_0 0.0 10.885",
+            "lbi-1272-135031-0000-B-1": "lbi-1272-135031-0000_1 0.0 10.885",
         },
     }
 
 
-@pytest.mark.xfail(reason="multi file recordings not supported yet")
```
Comment (desh2608) [inline test/test_kaldi_dirs.py:129]: @jtrmal This multi-channel recording test which was expected to fail earlier is now passing.

**Comment by popcornell** [comment]
@desh2608 On my side everything seems to work right now.

**Comment by pzelasko** [comment]
Thanks @popcornell and @desh2608, merging!

**Pull request merged**